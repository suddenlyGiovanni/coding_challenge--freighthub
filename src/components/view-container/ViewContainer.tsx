import * as React from 'react'
import styled from '@emotion/styled/macro'

import Typography from '@material-ui/core/Typography'
import Container from '@material-ui/core/Container'

const Main = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;

  width: 100%;
  padding-top: 30px;
`

export const ViewContainer: React.FC<{ title: string }> = ({
  children,
  title,
}) => {
  return (
    <Main>
      <Typography variant="h4" component="h2">
        {title}
      </Typography>
      <Container maxWidth={'md'}>{children}</Container>
    </Main>
  )
}
