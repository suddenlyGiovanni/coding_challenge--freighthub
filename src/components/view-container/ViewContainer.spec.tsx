import React from 'react'
import { render } from '@testing-library/react'

import { ViewContainer } from './ViewContainer'

describe('ViewContainer', () => {
  test('snapshot', () => {
    // Arrange
    const ChildrenNode = () => <div />
    const title = 'title'
    const { container } = render(
      <ViewContainer title={title}>
        <ChildrenNode />
      </ViewContainer>
    )

    // Assert
    expect(container).toMatchSnapshot()
  })
})
