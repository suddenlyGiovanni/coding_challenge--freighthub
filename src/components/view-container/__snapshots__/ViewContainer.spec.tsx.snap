// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewContainer snapshot 1`] = `
<div>
  <div
    class="css-ei4r1f-Main e1n1cmjm0"
  >
    <h2
      class="MuiTypography-root MuiTypography-h4"
    >
      title
    </h2>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <div />
    </div>
  </div>
</div>
`;
